crates/hir_def/src/item_tree.rs

```diff
@@ -501,6 +501,7 @@ pub struct Function {
     pub visibility: RawVisibilityId,
     pub generic_params: GenericParamsId,
     pub has_self_param: bool,
+    pub has_body: bool,
```

Comment:
I think this is probably OK, but we should consider refatoring the `body` query to just return an `Option` instead.